app/routes/Router.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace routes;
+
+use Exception;
+
+final class Router
+{
+    public array $routes = [];
+
+    public string $uri;
+
+    public bool $match = false;
+
+    public function __construct()
+    {
+        $this->routes = [
+            '/' => ['controller' => 'Home', 'action' => 'index'],
+            '/about' => ['controller' => 'About', 'action' => 'index'],
+            '/contact' => ['controller' => 'Contact', 'action' => 'index'],
+            '/product' => ['controller' => 'Product', 'action' => 'index'],
+            '/product/(?P\d+)' => ['controller' => 'Product', 'action' => 'show', 'params' => ['id' => ':id']],
+        ];
+
+        $this->uri = parse_url($_SERVER['REQUEST_URI'], PHP_URL_PATH);
```

Comment:
can be passed as an argument to the Router constructor from the index.php file

Reply:
changed